include/dynamatic/Dialect/Handshake/HandshakeTypes.td

```diff
@@ -341,17 +341,30 @@ class AllDataTypesMatchWithVariadic<string variadic, list<string> nonvariadics>
   >.result # ")">
 >;
 
+/// Multi-entity constraint ensuring that all HandshakeTypes including variadic
+/// have matching extra signals.
+/// The core implementation of this predicate is in C++.
+class AllExtraSignalsMatchWithVariadic<string variadic, list<string> nonvariadics> : PredOpTrait<
+  "all of {" # variadic # ", " # !interleave(nonvariadics, ", ") # "} should "
+  "have the same extra signals except for " # except,
+  CPred<"::dynamatic::handshake::doesExtraSignalsMatch(" #
+    concatArrays<"llvm::ArrayRef<ExtraSignal>",
+    variadicToExtraSignalArrays<variadic>.result,
+    valuesToExtraSignalArrays<nonvariadics>.result
+  >.result # ")">
+>;
+
 /// Multi-entity constraint ensuring that all HandshakeTypes including variadic
 /// have matching extra signals, except for any signal with the given `except` name.
 /// The core implementation of this predicate is in C++.
 class AllExtraSignalsMatchWithVariadicExcept<string except, string variadic, list<string> nonvariadics> : PredOpTrait<
```

Comment:
This trait is unused, but I think to have the complete set of

- `AllDataTypesMatch`
- `AllExtraSignalsMatch`
- `AllExtraSignalsMatchExcept`
- `AllDataTypesMatchWithVariadic`
- `AllExtraSignalsMatchWithVariadic`
- `AllExtraSignalsMatchExceptWithVariadic`

is preferable? I also want to add `AllTypesMatchWithVariadic`, corresponding to the built-in `AllTypesMatch`.

Reply:
Yeah, and on second thought people may use the built-in `SameOperandsAndResultType` in most cases (which also simplify the builder method and IR).
So I want to leave them as-is